lib/file.js

```diff
@@ -8,207 +8,143 @@
-
-function writeBinaryFile(FN, DATA) {
-    var BinaryStream = CreateObject("ADODB.Stream");
-    BinaryStream.Type = PipeIPC.adTypeBinary;
-    BinaryStream.Open();
-    BinaryStream.Write(DATA);
-    BinaryStream.SaveToFile(FN, adSaveCreateOverWrite);
-    BinaryStream.Close();
+function writeBinaryFile(path, data) {
+    var binaryStream = CreateObject("ADODB.Stream");
+    binaryStream.Type = PipeIPC.adTypeBinary;
+    binaryStream.Open();
+    binaryStream.Write(data);
+    binaryStream.SaveToFile(path, adSaveCreateOverWrite);
+    binaryStream.Close();
 }
 
-/////////////////////////////////////////////////////////////////////////////////
-// moveFile
-/////////////////////////////////////////////////////////////////////////////////
-
-function moveFile(FROM, TO) {
-    var FSO = CreateObject("Scripting.FileSystemObject");
-    var result = FSO.MoveFile(FROM, TO);
-    FSO = null;
-    return result;
+function moveFile(fromPath, toPath) {
+    return UseObject("Scripting.FileSystemObject", function(fso) {
+        return fso.MoveFile(fromPath, toPath);
+    });
 }
 
-/////////////////////////////////////////////////////////////////////////////////
-// createFolder
-/////////////////////////////////////////////////////////////////////////////////
-
-function createFolder(FN) {
-    var FSO = CreateObject("Scripting.FileSystemObject");
-    var result = FSO.CreateFolder(FN);
-    FSO = null;
-    return result;
+function createFolder(path) {
+    return UseObject("Scripting.FileSystemObject", function(fso) {
+        return fso.CreateFolder(path);
+    });
 }
 
-/////////////////////////////////////////////////////////////////////////////////
-// deleteFolder
-/////////////////////////////////////////////////////////////////////////////////
-
-function deleteFolder(FN) {
-    var FSO = CreateObject("Scripting.FileSystemObject");
-    var result = FSO.DeleteFolder(FN);
-    FSO = null;
-    return result;
+function deleteFolder(path) {
+    return UseObject("Scripting.FileSystemObject", function(fso) {
+        return fso.DeleteFolder(path);
+    });
 }
 
-/////////////////////////////////////////////////////////////////////////////////
-// deleteFile
-/////////////////////////////////////////////////////////////////////////////////
-
-function deleteFile(FN) {
-    var FSO = CreateObject("Scripting.FileSystemObject");
-    var result = FSO.DeleteFile(FN);
-    FSO = null;
-    return result;
+function deleteFile(path) {
+    return UseObject("Scripting.FileSystemObject", function(fso) {
+        return fso.DeleteFile(path);
+    });
 }
 
-/////////////////////////////////////////////////////////////////////////////////
-// includeFile
-/////////////////////////////////////////////////////////////////////////////////
-
-function includeFile(FN) {
+function includeFile(path) {
     try {
-        eval(readFile(FN));
+        eval(readFile(path));
     } catch (e) {
-        console.error(e.message, "in", FN);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Log Injection occurs when untrusted user input is directly written to log files without proper sanitization. This can allow attackers to manipulate log entries, potentially leading to security issues like log forging or cross-site scripting. To prevent this, always sanitize user input using `encodeURIComponent()` or `DOMPurify.sanitize()` before logging.  Learn more - https://cwe.mitre.org/data/definitions/117.html